src/Corax/Queries/BinaryMatch.Boolean.cs

```diff
@@ -22,24 +22,74 @@ static int FillFunc(ref BinaryMatch<TInner, TOuter> match, Span<long> matches)
 
                 while (true)
                 {
-                    var results = inner.Fill(matches);
-                    if (results == 0)
+                    int totalResults = 0;
+                    int iterations = 0;
+
+                    // PERF: An alternative implementation would be to perform OR in place. The upside is that every improvement on
+                    //       OR would impact everywhere this happens, but the vectorized Sort may also tip the balance here. Another
+                    //       good behavior would be that in cases of many duplicates we will have a better use of the buffer because
+                    //       OR will also deduplicate on every call. 
+
+                    var resultsSpan = matches;
+                    while (resultsSpan.Length > 0)
+                    {
+                        // RavenDB-17750: We have to fill everything possible UNTIL there are no more matches availables.
```

Comment:
what if there are more matches than there is space for them?

Reply:
It will continue from there. This is the key difference between `Fill` and `AndWith`